File: 0257-binary-tree-paths/0257-binary-tree-paths.cpp
class Solution {
public:
    vector<string> dfs(TreeNode*& currNode, vector<string>& paths, string path) {
        if (!currNode->left && !currNode->right) {
            if (find(paths.begin(), paths.end(), path) == paths.end()) {
                paths.push_back(path);
            }
            return paths;
        }

        if (currNode->left) {
            paths = dfs(currNode->left, paths, path+"->"+to_string(currNode->left->val));
        }
        if (currNode->right) {
            paths = dfs(currNode->right, paths, path+"->"+to_string(currNode->right->val));
        }

        return paths;
    }
    vector<string> binaryTreePaths(TreeNode* root) {
        if (!root) return {};
        vector<string> paths;
        paths = dfs(root, paths, to_string(root->val));
        return paths;
    }
};

File: 0257-binary-tree-paths/0257-binary-tree-paths.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def binaryTreePaths(self, root: Optional[TreeNode]) -> List[str]:
        paths = []
        def dfs(node, path):
            if not node.left and not node.right:
                if path not in paths:
                    paths.append(path)
            if node.left:
                dfs(node.left, path + "->"+ str(node.left.val))
            if node.right:
                dfs(node.right, path + "->"+ str(node.right.val))
        dfs(root, str(root.val))
        return paths